### ISSUE TITLE
Bind array parameter in query string.

### ISSUE THREAD

**Issue by PMExtra** [aspnet/Mvc#6701]
In some front-end AJAX library, the array parameter in GET query will be serialized to:

```
http://url?param[]=1&param[]=2
```

It's generated by some query like `ajax.get("url", { param: [ 1, 2 ] })`

There is a different with AspNetCore.Mvc standard. Mvc standard define the action by `Get([FromQuery] int[] param)` and query by `param=1&param=2`.

So why don't we support the parameter name contains `[]` ?

**Comment by pranavkm** [comment]
cc @Tratcher

**Comment by Tratcher** [comment]
Query parses it like that. Is the binder not working?

**Comment by pranavkm** [comment]
The value provider would look for `param` and not `param[]`. Is that the key the query parser stores the values with?

**Comment by Tratcher** [comment]
param is the key, but the value is a StringValue with multiple entries.

**Comment by pranavkm** [comment]
@Tratcher I tried this out and it looks like the key is `param[]` and not `param`. 

@PMExtra we do normalize JQuery's wire format but that only works for POST requests. Switching to `ajax.post` and updating your action to `HttpPost` should fix this. Alternatively you can look at https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/JQueryFormValueProviderFactory.cs and adapt it for GET requests. Normalizing these tokens is expensive both in terms of processing and allocations so it's unlikely we would have an adapter for query strings out of the box.

**Comment by Tratcher** [comment]
`param[]=1&param[]=2`?

**Comment by pranavkm** [comment]
Spoke to @Tratcher offline. Query parser doesn't do any work to unwrap this. My suggestions in the previous comment should get this sorted.

**Comment by rynowak** [comment]
We're already planning to address this for 2.1 by adding a query string provider for this that does the same thing as the form data one https://github.com/aspnet/Mvc/issues/6372

**Comment by rynowak** [comment]
Closing this out as a dupe. We'll use #6372 to track the work.
